code/modules/mob/living/simple_animal/bot/secbot.dm

```diff
@@ -404,17 +411,34 @@
 /mob/living/simple_animal/bot/secbot/explode()
 	visible_message("<span class='boldannounce'>[src] blows apart!</span>")
 	var/atom/Tsec = drop_location()
-
-	var/obj/item/bot_assembly/secbot/Sa = new (Tsec)
-	Sa.build_step = 1
-	Sa.add_overlay("hs_hole")
-	Sa.created_name = name
-	new /obj/item/assembly/prox_sensor(Tsec)
-	if(!noloot)
+	if(ranged)
+		var/obj/item/bot_assembly/ed209/Sa = new (Tsec)
+		Sa.build_step = 1
+		Sa.add_overlay("hs_hole")
+		Sa.created_name = name
+		new /obj/item/assembly/prox_sensor(Tsec)
+		var/obj/item/gun/energy/disabler/G = new (Tsec)
+		G.cell.charge = 0
+		G.update_icon()
+		if(prob(50))
+			new /obj/item/bodypart/l_leg/robot(Tsec)
+			if(prob(25))
+				new /obj/item/bodypart/r_leg/robot(Tsec)
+		if(prob(25))//50% chance for a helmet OR vest
+			if(prob(50))
+				new /obj/item/clothing/head/helmet(Tsec)
+			else
+				new /obj/item/clothing/suit/armor/vest(Tsec)
+	else
+		var/obj/item/bot_assembly/secbot/Sa = new (Tsec)
+		Sa.build_step = 1
+		Sa.add_overlay("hs_hole")
+		Sa.created_name = name
+		new /obj/item/assembly/prox_sensor(Tsec)
 		drop_part(baton_type, Tsec)
 
-	if(prob(50))
-		drop_part(robot_arm, Tsec)
+		if(prob(50))
+			drop_part(robot_arm, Tsec)
```

Comment:
Right now this proc is exactly like the code you're aiming to fix with this PR, and I question why code very specific to ED209, not potential future ranged bots, is being defined in the parent type. 

If component parts are stored in a list, most of this proc could be simplified to 

```
for(var/obj/item/I in component_parts)
    if(prob(50))
        drop_part(I, loc)
```

And this also future-proofs it against later bots.

Reply:
Alright, I think ive cleaned it to a reasonable degree that the ranged var is no longer an issue.



> If component parts are stored in a list, most of this proc could be simplified to this to make it drop an assortment of random parts

Good Idea. This PR is getting a bit too big for my liking however. I'll keep it in mind for the next pr